### DISCUSSION TITLE
issue: useController / Controller : Bad value returned

### DISCUSSION THREAD

**Discussion by squiaios** [react-hook-form/react-hook-form#7480] [Q&A]
### Version Number

7.22.5

### Codesandbox/Expo snack

https://codesandbox.io/s/wispy-mountain-cmh3v

### Steps to reproduce

Hello folks,

I have a bug (or I missing something) about `useController`. The value returned is not good

1. Initialize form with 2 attributes with empty string
2. when component is mounted, I reset these 2 fields (in my case I receive data from external source)
3. I change the locale
4. The value returned from useController is not good

PS : None of my component is using `React.memo`

Thank you very much for your help :)

### Expected behaviour

I expect that the value returned by getValues to be exact.

However, using `getValues()` from `useFormContext` return the good data

### What browsers are you seeing the problem on?

Chrome

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by bluebill1049** [comment]
I don't get the issue in here: what's ` Bad value returned`? and what's the expected result?

**Comment by squiaios** [comment]
If the local used is English, I expect to have "Hello" in input, but if I
change the locale to French, I expect to have "Bonjour" in input

**Comment by bluebill1049** [comment]
you are using `setState` instead of `setValue` (https://react-hook-form.com/api/useform/setvalue), `setState` has no connecting with react hook form.

**Comment by squiaios** [comment]
I'm using `setState` to change the locale, but I do not need to "save" the locale value in my form.
It's only for avoiding to display all inputs in same time, just the one for the locale choosen

**Reply by bluebill1049** [reply]
if you want to update input value, use setValue, you can use use the value returned by useController and display the data

**Comment by squiaios** [comment]
`onChange` function from `useController` works well, it's just the value
returned by `useController` after the locale is changed that is not updated.

When I change the locale, `FormGroup` component is re-render, `nameLocalized` is the new name with the locale, but somewhere in
useController (I did not found where), do not take the value for the key nameLocalized (in my case `title:en` or `title:fr`).
